cmd/podspec-to-arm/mock_aci.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the Apache 2.0 license.
+*/
+package main
+
+import (
+	"context"
+
+	azaciv2 "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerinstance/armcontainerinstance/v2"
+	"github.com/virtual-kubelet/virtual-kubelet/node/api"
+)
+
+type CreateContainerGroupFunc func(ctx context.Context, resourceGroup, podNS, podName string, cg *azaciv2.ContainerGroup) error
+type GetContainerGroupInfoFunc func(ctx context.Context, resourceGroup, namespace, name, nodeName string) (*azaciv2.ContainerGroup, error)
+type GetContainerGroupListFunc func(ctx context.Context, resourceGroup string) ([]*azaciv2.ContainerGroup, error)
+type ListCapabilitiesFunc func(ctx context.Context, region string) ([]*azaciv2.Capabilities, error)
+type DeleteContainerGroupFunc func(ctx context.Context, resourceGroup, cgName string) error
+type ListLogsFunc func(ctx context.Context, resourceGroup, cgName, containerName string, opts api.ContainerLogOpts) (*string, error)
+type ExecuteContainerCommandFunc func(ctx context.Context, resourceGroup, cgName, containerName string, containerReq azaciv2.ContainerExecRequest) (*azaciv2.ContainerExecResponse, error)
+
+type GetContainerGroupFunc func(ctx context.Context, resourceGroup, containerGroupName string) (*azaciv2.ContainerGroup, error)
+
+type MockACIProvider struct {
+	MockCreateContainerGroup    CreateContainerGroupFunc
+	MockGetContainerGroupInfo   GetContainerGroupInfoFunc
+	MockGetContainerGroupList   GetContainerGroupListFunc
+	MockListCapabilities        ListCapabilitiesFunc
+	MockDeleteContainerGroup    DeleteContainerGroupFunc
+	MockListLogs                ListLogsFunc
+	MockExecuteContainerCommand ExecuteContainerCommandFunc
+
+	MockGetContainerGroup GetContainerGroupFunc
+}
+
+func NewMockACIProvider(capList ListCapabilitiesFunc) *MockACIProvider {
+	mock := &MockACIProvider{}
+	mock.MockListCapabilities = capList
+	return mock
+}
+
+func (m *MockACIProvider) ListCapabilities(ctx context.Context, region string) ([]*azaciv2.Capabilities, error) {
+	if m.MockListCapabilities != nil {
+		return m.MockListCapabilities(ctx, region)
+	}
+	return nil, nil
```

Comment:
the second nil could be replaced with error messages (same comment for the other funcs below)